### PR TITLE
resist the bitrot!

### PR DESCRIPTION
**Pull request by ewollesen**
Changes to shake the rust out of this package, and make sure it's properly built and tested.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/tidepool-org/go-common
 
-go 1.14
+go 1.18
```
Comment (toddkazakov) [inline go.mod:3]: Why not `1.21`?

Comment (ewollesen) [reply go.mod:3]: My understanding of that value, is that it represents the minimum go toolchain version that is required to build/use the module. [Reference](https://go.dev/ref/mod#go-mod-file-go)

So 1.18 felt the most honest in that respect.

**Comment by ewollesen** [comment]
Alex landed a bunch of changes recently that removed the need for a couple of these changes. I've rebased and force pushed to bring in his changes. In the process, a couple of commits were dropped.

**Review by Roukoswarf — APPROVED** [review]
LGTM

**Pull request merged**